core/src/main/java/org/aerogear/mobile/core/configuration/https/HttpsConfiguration.java

```diff
@@ -0,0 +1,54 @@
+package org.aerogear.mobile.core.configuration.https;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This represents a parsed Https configuration from JSON configuration
+ */
+public class HttpsConfiguration {
+    public static final String CERT_PINNING_KEY = "certificatePins";
+
+    private final List<CertificatePinningEntry> certPinningConfig;
+
+    private HttpsConfiguration(final List<CertificatePinningEntry> certPinningConfig) {
+        this.certPinningConfig = certPinningConfig;
+    }
+
+    public static class Builder {
+        private List<CertificatePinningEntry> certPinningConfig = new ArrayList<>();
+
+        public Builder setCertPinningConfig(final List<CertificatePinningEntry> pinningConfig) {
+            this.certPinningConfig = pinningConfig;
```

Comment:
The list should be copied to avoid having reference to mutable internal exposed to the caller.